libs/langchain-google-cloud-sql-pg/src/DocumentLoader.ts

```diff
@@ -60,10 +89,72 @@ export class PostgresLoader extends BaseDocumentLoader {
         query = `SELECT * FROM "${schemaName}"."${tableName}"`;
       }
 
-      return new PostgresLoader(engine, options);
+      try {
+        const result = await engine.pool.raw(query);
+        const columnNames = result.fields.map((field: { name: any; }) => field.name);
+
+        contentColumns = contentColumns || [columnNames[0]];
+        metadataColumns = metadataColumns || columnNames.filter((col: string) => !contentColumns?.includes(col));
+
+        if (metadataJsonColumn && !columnNames.includes(metadataJsonColumn)) {
+            throw new Error(`Column ${metadataJsonColumn} not found in query result ${columnNames}.`);
+        }
+
+        if (!metadataJsonColumn && columnNames.includes(DEFAULT_METADATA_COL)) {
+            metadataJsonColumn = DEFAULT_METADATA_COL;
+        }
+
+        const allNames = [...(contentColumns || []), ...(metadataColumns || [])];
+        allNames.forEach(name => {
+            if (!columnNames.includes(name)) {
+                throw new Error(`Column ${name} not found in query result ${columnNames}.`);
+            }
+        });
+
+        return new PostgresLoader(engine, options);
```

Comment:
Instead of setting options as a Loader property, let's just set the variables we need to save like engine, query, content, metadata, and metadata json columns, and formatter